### PR TITLE
Hours screen

### PR DESCRIPTION
**Pull request by samanthalai0914**
Display hours and restock indicators on mobile app.

Also some changes on home screen, sorry. Cosmetic only.

### CODE REVIEW

File: screens/HoursScreen.js [hunk 1]
```diff
@@ -0,0 +1,152 @@
...
+        });
+        this.setState({
+          daysToHours: daysToHours
+        });
+      }.bind(this));
+  }
+
+  getEmojis(restock) {
+    let arr = new Array();
+    Object.keys(EMOJIS).map(function(key, emoji){
```
Comment (wayne-li2) [inline screens/HoursScreen.js]: Style: space before the "{"

File: screens/HoursScreen.js [hunk 2]
```diff
@@ -0,0 +1,152 @@
...
+      if (restock[key] == 1){
```
Comment (wayne-li2) [inline screens/HoursScreen.js]: Sylte: space before the "{"

File: screens/HoursScreen.js (hunk 1 above)
Comment (wayne-li2) [inline screens/HoursScreen.js]: Also, looks like "emoji" variable isn't used. You can just have function(key) and that will be fine.

File: screens/HoursScreen.js [hunk 3]
```diff
@@ -0,0 +1,152 @@
...
+          }
+        });
+        this.setState({
+          daysToHours: daysToHours
+        });
+      }.bind(this));
+  }
+
+  getEmojis(restock) {
+    let arr = new Array();
```
Comment (wayne-li2) [inline screens/HoursScreen.js]: I think we can simplify this method.

For example:
If I have:
`arr = [1, 2, 3]`

I can do:

`return arr.map(function(element) {
  return element * 2
})`

and this method will return me:

`[2, 4, 6]`

Without needing to call "push". I realize this is implemented wrong in HomeScreen :) I only learned this today.

File: screens/HomeScreen.js [hunk 4]
```diff
@@ -63,11 +67,19 @@ export default class HomeScreen extends React.Component {
           key={categoryKey}
           style={styles.touchable}
           onPress={this.onPressCategory.bind(this, categoryKey, categoryDisplayName)}>
-            <ImageBackground
-              source={categoryImage}
+          <View style={styles.iconHolder}>
```
[the remark this answers is not in the document]
Comment (samanthalai0914) [reply screens/HomeScreen.js]: precaching, images, and constants file are gone 

TODO: modify DB schema

File: App.js [hunk 5]
```diff
@@ -8,7 +8,6 @@ import { Ionicons } from '@expo/vector-icons';
 
 import firebase from 'firebase';
```
Comment (wayne-li2) [inline App.js]: Can we delete this import as it's unused?

Can we also delete "import { Asset } from 'expo-asset';" as it is also unused?

File: constants/Icons.js [hunk 6]
```diff
@@ -0,0 +1,13 @@
+const EMOJIS = {
```
Comment (wayne-li2) [inline constants/Icons.js]: Can we rename this file to something specific to restock indicators? Maybe RestockIcons.js? With just Icons.js it's possible this is just generalized icons.

File: screens/HomeScreen.js [hunk 7]
```diff
@@ -8,11 +8,9 @@ import {
   TouchableOpacity,
```
Comment (wayne-li2) [inline screens/HomeScreen.js]: Can we remove "ImageBackground" from the react-native imports, it is unused.

File: screens/HomeScreen.js [hunk 8]
```diff
@@ -52,22 +49,25 @@ export default class HomeScreen extends React.Component {
       categoryDisplayName: displayName,
     });
   }
+
```
Comment (wayne-li2) [inline screens/HomeScreen.js]: Still need to delete this extra line here.

File: screens/HomeScreen.js [hunk 9]
```diff
@@ -96,7 +95,7 @@ const styles = StyleSheet.create({
     backgroundColor: '#fff',
```
Comment (wayne-li2) [inline screens/HomeScreen.js]: Looks like "win" constant isn't used. Let's delete it.

File: screens/HoursScreen.js [hunk 10]
```diff
@@ -0,0 +1,142 @@
...
+
+  renderHours() {
+    var orderedDaysToHours = Days.ORDERED_READABLE_DAYS.map((readableDay) => {
+      return this.state.daysToHours[readableDay];
+    });
+    return orderedDaysToHours.map((dayObject) => {
+      if (dayObject) {
+        return (
+        <View key={dayObject.day} style={styles.dayContainer}>
+          <Text style={styles.dayText} >{dayObject.day} </Text>
```
Comment (wayne-li2) [inline screens/HoursScreen.js]: Spacing for this line

File: screens/HoursScreen.js [hunk 11]
```diff
@@ -0,0 +1,142 @@
...
+          <View style={styles.hoursRestockContainer}>
+            <Text style={styles.hoursText}> {dayObject.hours} </Text>
```
Comment (wayne-li2) [inline screens/HoursScreen.js]: Spacing for this line

File: screens/HoursScreen.js [hunk 12]
```diff
@@ -0,0 +1,142 @@
...
+            <Text style={styles.hoursText}> {this.getEmojis(dayObject.restock)} </Text>
```
Comment (wayne-li2) [inline screens/HoursScreen.js]: Spacing for this line

File: screens/HoursScreen.js [hunk 13]
```diff
@@ -0,0 +1,142 @@
...
+    });
+  }
+
+  renderHours() {
+    var orderedDaysToHours = Days.ORDERED_READABLE_DAYS.map((readableDay) => {
+      return this.state.daysToHours[readableDay];
+    });
+    return orderedDaysToHours.map((dayObject) => {
+      if (dayObject) {
+        return (
```
Comment (wayne-li2) [inline screens/HoursScreen.js]: Are the parentheses necessarily? Not sure -- can you try without em?

Comment (samanthalai0914) [reply screens/HoursScreen.js]: necessary

**Pull request merged**